iiif_prep/iiif_prep/preparer.py

```diff
@@ -0,0 +1,153 @@
+import logging
+
+from configparser import ConfigParser
+from os import listdir
+from os.path import isdir, join
+from shutil import copytree
+
+from .archivesspace import ArchivesSpaceClient, NoResultsError, MultipleResultsError
+
+
+class Preparer():
+    CURRENT = 0
+    MICROFILM = 1
+    LEGACY = 2
+
+    def __init__(self):
+        logging.basicConfig(
+            datefmt='%m/%d/%Y %I:%M:%S %p',
+            filename='iiif_preparer.log',
+            format='%(asctime)s %(message)s',
+            level=logging.INFO)
+        self.config = ConfigParser()
+        self.config.read("local_settings.cfg")
+
+    def run(self, source_directory, target_directory):
+        as_client = ArchivesSpaceClient(
+            self.config.get("ArchivesSpace", "baseurl"),
+            self.config.get("ArchivesSpace", "username"),
+            self.config.get("ArchivesSpace", "password"),
+            self.config.get("ArchivesSpace", "repository"))
+        officers = sorted(
+            self.get_officers(
+                source_directory,
+                self.config.get(
+                    "IgnoreList",
+                    "ignore_list")))
+        for officer in officers:
+            self.officer_path = join(source_directory, officer)
+            self.structure = self.determine_structure(officer)
+            diaries = self.get_list_of_diaries()
+            logging.info("Starting {}. {} diaries to process.".format(
+                officer, len(diaries)))
+            for d in diaries:
+                logging.info("Starting {}".format(d))
+                self.mezzanine_directory = self.get_mezzanine_path(
+                    officer, d)
+                try:
+                    self.refid = as_client.get_diary_refid(d)
+                    logging.info(
+                        "Found ref_id {} for {}. Copying...".format(self.refid, d))
+                    self.copy_files(target_directory)
+                except NoResultsError as e:
+                    if self.structure == self.LEGACY:
+                        try:
+                            self.refid = as_client.get_diary_refid([f for f in listdir(
+                                join(self.officer_path, "Service Edited"))][0][:-4])
+                            logging.info(
+                                "Found ref_id {} for {}. Copying...".format(self.refid, d))
```

Comment:
```suggestion
                                "Found ref_id {} for diary {}. Copying...".format(self.refid, d))
```
Do you want to indicate this was found in a secondary lookup?